### PR TITLE
cec: support resource name qualification for HTTP HealthCheckFilter

### PR DESCRIPTION
**Pull request by mhofstetter**
Currently, Envoy resource names within a `CiliumEnvoyConfig` are qualified by prefixing them with the namespace and name of the `CiliumEnvoyConfig`. This affects the names itself - but also references from other Envoy resources.

This commit introduces support for qualifying the cluster references of the Envoy HTTP `HealthCheckFilter`.

### CODE REVIEW

File: pkg/ciliumenvoyconfig/cec_resource_parser.go [hunk 1]
```diff
@@ -553,6 +559,40 @@ func injectCiliumL7Filter(hcmConfig *envoy_config_http.HttpConnectionManager) bo
...
+					if nameUpdated {
+						updated = true
+						clusterName = updatedClusterName
+					}
+
+					clusters[clusterName] = p
+				}
+				httpFilterConfig.ClusterMinHealthyPercentages = clusters
+
+				h.TypedConfig = toAny(httpFilterConfig)
```
Comment (jrajahalme) [inline pkg/ciliumenvoyconfig/cec_resource_parser.go]: These two lines should only be executed if `updated == true`

**Review by jrajahalme — APPROVED** [review]
Minor nit on potentially modifying hcmConfig even when returned `updated` is `false`.

File: pkg/ciliumenvoyconfig/cec_resource_parser.go (hunk 1 above)
Comment (mhofstetter) [reply pkg/ciliumenvoyconfig/cec_resource_parser.go]: good point - thanks 

-> changed

**Pull request merged**